src/resolve/resolved-dns-browse-services.h

```diff
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+typedef struct DnsServiceBrowser DnsServiceBrowser;
+typedef struct DnssdDiscoveredService DnssdDiscoveredService;
+
+#include "sd-varlink.h"
+#include "resolved-dns-rr.h"
+#include "resolved-dns-question.h"
+#include "resolved-dns-answer.h"
+//#include "resolved-dns-query.h"
+
+typedef struct DnsQuery DnsQuery;
+typedef struct DnsScope DnsScope;
+typedef struct Manager Manager;
+typedef enum DnsRecordTTLState DnsRecordTTLState;
+
+enum DnsRecordTTLState {
+        DNS_RECORD_TTL_STATE_80_PERCENT,
+        DNS_RECORD_TTL_STATE_85_PERCENT,
+        DNS_RECORD_TTL_STATE_90_PERCENT,
+        DNS_RECORD_TTL_STATE_95_PERCENT,
+        _DNS_RECORD_TTL_STATE_MAX,
+        _DNS_RECORD_TTL_STATE_MAX_INVALID = -EINVAL
+};
+
+struct DnssdDiscoveredService {
+        unsigned n_ref;
+        DnsServiceBrowser *service_browser;
+        sd_event_source *schedule_event;
+        DnsResourceRecord *rr;
+        int family;
+        usec_t until;
+        DnsRecordTTLState rr_ttl_state;
+        DnsQuery *query;
+        LIST_FIELDS(DnssdDiscoveredService, dns_services);
+};
+
+struct DnsServiceBrowser {
+        unsigned n_ref;
+        Manager *manager;
+        sd_varlink *link;
+        DnsQuestion *question_idna;
+        DnsQuestion *question_utf8;
+        uint64_t flags;
+        sd_event_source *schedule_event;
+        usec_t delay;
+        DnsResourceKey *key;
+        int ifindex;
+        uint64_t token;
+        LIST_HEAD(DnssdDiscoveredService, dns_services);
+};
+
+usec_t mdns_calculate_next_query_delay(usec_t current_delay);
+usec_t mdns_maintenance_next_time(usec_t until, uint32_t ttl, DnsRecordTTLState ttl_state);
+usec_t mdns_maintenance_jitter(uint32_t ttl);
+
+DnsServiceBrowser *dns_service_browser_free(DnsServiceBrowser *sb);
+void dns_remove_service(DnsServiceBrowser *sb, DnssdDiscoveredService *service);
+DnssdDiscoveredService *dns_service_free(DnssdDiscoveredService *service);
+
+DnsServiceBrowser *dns_service_browser_ref(DnsServiceBrowser *sb);
+DnsServiceBrowser *dns_service_browser_unref(DnsServiceBrowser *sb);
+
+DnssdDiscoveredService *dns_service_ref(DnssdDiscoveredService *service);
+DnssdDiscoveredService *dns_service_unref(DnssdDiscoveredService *service);
```

Comment:
the types really should match the func name prefix.

i.e. if a type is called DnssdDiscoveredService the funcs should usually be called dnssd_discovered_service_xyz()

(exceptions of this rule are ok in many cases, but i don't see that here)